connectivity/check/deployment.go

```diff
@@ -1157,8 +1162,10 @@ func (ct *ConnectivityTest) validateDeployment(ctx context.Context) error {
 	}
 
 	for _, pod := range clientPods.Items {
-		if err := WaitForCiliumEndpoint(ctx, ct, ct.clients.src, ct.Params().TestNamespace, pod.Name); err != nil {
-			return err
+		if !disableEndpointCRD {
+			if err := WaitForCiliumEndpoint(ctx, ct, ct.clients.src, ct.Params().TestNamespace, pod.Name); err != nil {
```

Comment:
Is the intent here to better support testing Cilium that's configured in kvstore mode?

I'm tracing this back to https://github.com/cilium/cilium/pull/5933 , which given the timing may be when we were initially introducing CEP support so most users were on kvstore.

The main concern I have with this approach is the potential for race conditions. Currently, this check seems to ensure that some state is configured in the cluster before continuing. If we skip that check entirely and don't create an equivalent check for kvstore state, then this could cause tests to pass even if the cluster is not ready (because Cilium just isn't handling the traffic due to the race), or cause failures because the test didn't wait long enough (even if the test would otherwise pass if it _did_ wait long enough).

It's not obvious to me why this PR proposal is safe and why it won't introduce flakiness into the tests in this deployment scenario.

Reply:
I discussed this a bit with @marseel out-of-band and he brought up that these deployments are deployed by the CLI tool as part of `SetupAndValidate()`, explicitly *after* ensuring that Cilium is deployed correctly, so there's no pattern here where the apps are deployed without Cilium being set up correctly. As long as the Deployment is ready, this should provide a strong enough guarantee that the CEPs (or equivalent state) is also propagated.

In that case I have no remaining reservations. The main question to me would be whether we need this option at all - would it be simpler to just remove these wait conditions altogether instead of making them conditional?

Reply:
Thank you, Joe!
I don't have a strong opinion here... Maybe let's keep the condition at least for a while.

Reply:
The cost of "keeping it for a while" is that there's no obvious time to revisit this. By deleting it in only one specific test scenario, it makes it more likely we won't notice problems, and in the mean time the code is more complex. I'd rather keep the code simpler and remove it entirely if it's unnecessary. Then if it turns out we're wrong and it's needed, then we will also learn about that sooner and in all testing cases instead of just in this one new testing mode.

Reply:
Agree, I've removed the function `WaitForCiliumEndpoint`.

CC: @asauber @nathanjsweet